src/core/regimes.rkt

```diff
@@ -232,9 +232,9 @@
    ;; For each alt loop over its vector of errors
    (for ([alt-idx (in-naturals)] [alt-error-sums (in-vector flvec-psums)])
     ;; Loop over the points up to our current point
-    (for ([prev-split-idx (in-range 0 point-idx)])
-     ;; Check if we can add a split point
-     (when (vector-ref can-split-vec (+ prev-split-idx 1))
+    (for ([prev-split-idx (in-range 0 point-idx)]
+          [can-split-idx can-split-vec]
```

Comment:
This is wrong, you need to walk over `can-split-vec` offset by one. Also you've gotta use an `in-xxx` operation.

Reply:
Use `[can-split (in-vector can-split-vec 1)]`